scp/gen_operators/SCPOperatorSysGen.cpp

```diff
@@ -0,0 +1,270 @@
+/*
+* This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+* Distributed under the MIT License
+* (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+*/
+
+#include "scpUtils.hpp"
+#include "SCPOperatorSysGen.hpp"
+
+namespace scp
+{
+using namespace utils;
+SCPOperatorSysGen::SCPOperatorSysGen(const std::unique_ptr<ScMemoryContext> &ctx, ScAddr addr)
+  : SCPOperator(ctx, addr)
+{
+  operands = std::vector<SCPOperand*>(4);
+}
+
+std::string SCPOperatorSysGen::GetTypeName()
+{
+  return "sys_gen";
+}
+
+sc_result SCPOperatorSysGen::Parse()
```

Comment:
Looks like it is common method for several operators

Reply:
It is true. There is basic class